packages/map/src/components/UsaMap/components/UsaMap.State.tsx

```diff
@@ -118,6 +119,56 @@ const UsaMap = () => {
     }
   }, [data, state.general.territoriesAlwaysShow])
 
+  useEffect(() => {
+    unitedStates.map(geo => {
+      state.map.patterns.map((patternData, patternIndex) => {
+        const { pattern, dataKey, size } = patternData
+        let geoKey = geo.properties.iso
+        if (!geoKey) return
+        let legendColors
+        const geoData = data[geoKey]
+        if (geoData !== undefined) {
+          legendColors = applyLegendToRow(geoData)
+        }
+        const currentFill = legendColors[0]
+        const hasMatchingValues = patternData.dataValue === geoData[patternData.dataKey]
+        const patternColor = patternData.color || getContrastColor('#000', currentFill)
+
+        if (!hasMatchingValues) return
+        const passesContrastCheck = checkColorContrast(currentFill, patternColor)
+
+        setState(prevState => {
+          if (prevState.map.patterns[patternIndex].contrastCheck === passesContrastCheck) {
+            return prevState
+          }
+          return {
+            ...prevState,
+            runtime: {
+              ...state.runtime,
+              editorErrorMessage: !passesContrastCheck ? 'One or more patterns do not pass the WCAG 2.1 contrast ratio of 3:1.' : ''
+            },
+            map: {
+              ...prevState.map,
+              patterns: prevState.map.patterns.map((pattern, i) => {
+                if (i === patternIndex) {
+                  return {
+                    ...pattern,
+                    dataValue: geoData[patternData.dataKey],
+                    color: patternData.color,
+                    size: patternData.size,
+                    pattern: patternData.pattern,
+                    contrastCheck: passesContrastCheck
+                  }
+                }
+                return pattern
+              })
+            }
+          }
+        })
+      })
+    })
+  }, [state.map.patterns, data, state, setState, applyLegendToRow])
```

Comment:
this state.map.patterns dependency seems like a memory leak because of the set state you're calling directly from your useEffect.

once you setState here you have to rerun the useEffect. which leads me to believe that you already had everything you needed set the correct state in the setState context.

Reply:
In state.map.patterns we're checking if the previous state is different on state.map.patterns before calling the setState.

Reply:
I feel like you could just remove that from the dependency array and you wouldn't need that check.

Reply:
I've updated the useEffect so that the memory leak isn't happening anymore.